src/main/java/hudson/model/ExternalRun.java

```diff
@@ -116,23 +123,28 @@ private String elementText(XMLStreamReader r) throws XMLStreamException {
             }
 
             public Result run(BuildListener listener) throws Exception {
-                PrintStream logger = new PrintStream(new DecodingStream(listener.getLogger()));
+                PrintStream logger = new PrintStream(new DecodingStream(listener.getLogger()), false, "UTF-8");
 
                 XMLInputFactory xif = XMLInputFactory.newInstance();
                 XMLStreamReader p = xif.createXMLStreamReader(in);
 
                 p.nextTag();    // get to the <run>
                 p.nextTag();    // get to the <log>
 
-                setCharset(p.getAttributeValue(null,"content-encoding"));
-                while(p.next()!= END_ELEMENT) {
-                    int type = p.getEventType();
-                    if(type== CHARACTERS || type== CDATA)
-                        logger.print(p.getText());
+                try {
+                    setCharset(p.getAttributeValue(null,"content-encoding"));
+                    while(p.next()!= END_ELEMENT) {
+                        int type = p.getEventType();
+                        if(type== CHARACTERS || type== CDATA)
+                            logger.print(p.getText());
+                    }
+                    p.nextTag(); // get to <result>
+                } catch (Exception ex) {
+                    logger.print("An error occurred during encoding: " + ex.getMessage());
+                    throw ex;
```

Comment:
If you are throwing up the exception anyway, why are you bothering to print a message (which without a stack trace will often be meaningless anyway)?